internal/connectors/engine/engine.go

```diff
@@ -274,6 +277,35 @@ func (e *engine) CreateFormanceAccount(ctx context.Context, account models.Accou
 	return nil
 }
 
+func (e *engine) CreateFormancePayment(ctx context.Context, payment models.Payment) error {
+	run, err := e.temporalClient.ExecuteWorkflow(
+		ctx,
+		client.StartWorkflowOptions{
+			ID:                                       fmt.Sprintf("create-formance-payment-%s-%s", payment.ConnectorID.String(), payment.Reference),
+			TaskQueue:                                payment.ConnectorID.String(),
+			WorkflowIDReusePolicy:                    enums.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE,
+			WorkflowExecutionErrorWhenAlreadyStarted: false,
+			SearchAttributes: map[string]interface{}{
+				workflow.SearchAttributeStack: e.stack,
+			},
+		},
+		workflow.RunCreateFormancePayment,
+		workflow.CreateFormancePayment{
+			Payment: payment,
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	// Wait for bank account creation to complete
+	if err := run.Get(ctx, nil); err != nil {
+		return err
+	}
```

Comment:
This also has the potential to retry forever and leave the API hanging. Let's talk to Clem soon about this so we can come up with a retry policy that is sensible when thinking about APIs.

We might also want to consider eventual consistency. As in we do all the validation in the http handler, generate a unique ID and respond with HTTP Status Accepted (202) responding with only the unique ID in the response body.
Then whether the payment was created successfully or not can be tracked by polling a different API using the unique ID. That way the user doesn't have to wait for temporal retries to receive a "ok your payment request was submitted". A bit more user friendly and allowing us to have reasonable response times while allowing for temporal to do its thing in the background.

Reply:
Tickets created, we will discuss about that with the product team asap